---
title: "Lab 9-10: Structural Bioinformatics"
author: "Alex Cagle"
date: "2023-05-09"
output: pdf_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

## PDB Statistics:

PDB is the main database for structural information on proteins.

```{r}
db <- read.csv("PDB.csv")
head(db)
```

> Q1: What percentage of structures in the PDB are solved by X-Ray and Electron Microscopy.

```{r}
sum(as.numeric(gsub(",", "", db$X.ray)))
```

```{r}
em.total <- sum(as.numeric(gsub(",", "", db$EM)))
em.total
```


```{r}
# Writing function
sum_comma <- function(x) {
  # substitute comma and convert to numeric
  sum(as.numeric (gsub( ",", "", x)))
}

sum_comma(db$X.ray) / sum_comma(db$Total)
```

For EM:
```{r}
round(sum_comma(db$EM) / sum_comma(db$Total), 3)
```


> Q2: What proportion of structures in the PDB are protein?

```{r}
round(sum_comma(db$Total[1]) / sum_comma(db$Total), 2)
```


> Q3: Type HIV in the PDB website search box on the home page and determine how many HIV-1 protease structures are in the current PDB?

N/A

> Q4. Water molecules normally have 3 atoms. Why do we see just one atom per water molecule in this structure?

The resolution of the structure is too low to be able to see H atoms.

> Q5. There is a critical "conserved" water molecule in the binding site. Can you identify this water molecule? What residue number does this water moelcule have?

HOH308

## Working with structures in R

We can use the bio3d package to read and perform bioinformatics calculations on PDB structures.

```{r}
library(bio3d)

pdb <- read.pdb("1hsg")

pdb
```
```{r}
attributes(pdb)
```
```{r}
head(pdb$atom)
```
> Q7. How many amino acid residues are there in this pdb object?

198

> Q8. Name one of the two non-protein residues?

Mk1 and water

> Q9. How many protein chains are in this structure?

2

## Predicting function motions

```{r}
adk <- read.pdb("6s36")
adk
```
Perform a prediction of flexibility with a technique called NMA (normal mode analysis).

```{r}
m <- nma(adk)
```
```{r}
plot(m)
```
Write out a 'movie' (trajectory) of the motion for viewing in Molstar.

```{r}
mktrj(m, file="adk_m7.pdb")
```


```{r}
# Install packages in the R console NOT your Rmd/Quarto file

#install.packages("bio3d")
#install.packages("devtools")
#install.packages("BiocManager")

#BiocManager::install("msa")
#devtools::install_bitbucket("Grantlab/bio3d-view")
```

> Q10. Which of the packages above is found only on BioConductor and not CRAN?

**msa**

> Q11. Which of the above packages is not found on BioConductor on CRAN?

**bio3d-view**

> Q12. True or False? Functions from the devtools package can be used to install packages from GitHub and BitBucket?

**True**

```{r}
library(bio3d)

aa <- get.seq("1ake_A")
aa
```

> Q13. How many amino acids are in this sequence, i.e. how long is this sequence?

**214 amino acids long**

```{r}
# Blast search
# b <- blast.pdb(aa)
```

```{r}
# hits <- plot(b)
```

```{r}
# Checking what's in the hits object
# hits$pdb.id
```
```{r}
hits <- NULL
hits$pdb.id <- c('1AKE_A','6S36_A','6RZE_A','3HPR_A','1E4V_A','5EJE_A','1E4Y_A','3X2S_A','6HAP_A','6HAM_A','4K46_A','3GMT_A','4PZL_A')
```
```{r}
# Download related PDB files
files <- get.pdb(hits$pdb.id, path="pdbs", split=TRUE, gzip=TRUE)
```

```{r}
# Align related PDBs
pdbs <- pdbaln(files, fit = TRUE, exefile="msa")
```

```{r}
# Looking at the 'pdbs' object
pdbs
```

```{r}
# Now that we have our aligned and superposed structures, we can perform all sorts of analyses on them
# Let's do PCA:
pc.xray <- pca(pdbs)
pc.xray

plot(pc.xray)
```

```{r}
# Vector containing PDB codes for figure axis
ids <- basename.pdb(pdbs$id)

# Draw schematic alignment
plot(pdbs, labels=ids)
```

```{r}
# Results of PCA on Adenylate kinase X-ray structures:

# Calculate RMSD
rd <- rmsd(pdbs)

# Structure-based clustering
hc.rd <- hclust(dist(rd))
grps.rd <- cutree(hc.rd, k=3)

plot(pc.xray, 1:2, col="grey50", bg=grps.rd, pch=21, cex=1)
#plot(pc.xray, 1:2, col=grps.rd)
```

```{r}
# Visualize first principal component
pc1 <- mktrj(pc.xray, pc=1, file="pc_1.pdb")
pc1
```

